redaxo/src/addons/mediapool/lib/mediapool.php

```diff
@@ -0,0 +1,200 @@
+                $clang = $art_arr['clang_id'];
+                $ooa = rex_article::get($aid, $clang);
+                $name = ($ooa) ? $ooa->getName() : '';
+                $warning[0] .= '<li><a href="javascript:openPage(\'' . rex_url::backendPage('content', ['article_id' => $aid, 'mode' => 'edit', 'clang' => $clang]) . '\')">' . $name . '</a></li>';
+            }
+            $warning[0] .= '</ul>';
+        }
+
+        // ----- EXTENSION POINT
+        $warning = rex_extension::registerPoint(new rex_extension_point('MEDIA_IS_IN_USE', $warning, [
+            'filename' => $filename,
+        ]));
+
+        if (!empty($warning)) {
+            return implode('<br />', $warning);
+        }
+
+        return false;
+    }
+
+    /**
+     * check if mediatpye(extension) is allowed for upload.
+     *
+     * @param string $filename
+     *
+     * @return bool
+     */
+    public static function isAllowedMediaType($filename, array $args = [])
+    {
+        $fileExt = mb_strtolower(rex_file::extension($filename));
+
+        if ('' === $filename || str_contains($fileExt, ' ') || '' === $fileExt) {
+            return false;
+        }
+
+        if (str_starts_with($fileExt, 'php')) {
+            return false;
+        }
+
+        $blacklist = self::getMediaTypeBlacklist();
+        foreach ($blacklist as $blackExtension) {
+            // blacklisted extensions are not allowed within filenames, to prevent double extension vulnerabilities:
+            // -> some webspaces execute files named file.php.txt as php
+            if (str_contains($filename, '.'. $blackExtension)) {
+                return false;
+            }
+        }
+
+        $whitelist = self::getMediaTypeWhitelist($args);
+        return !count($whitelist) || in_array($fileExt, $whitelist);
+    }
+
+    /**
+     * Checks file against optional whitelist from property `allowed_mime_types`.
+     *
+     * @param string      $path     Path to the physical file
+     * @param null|string $filename Optional filename, will be used for extracting the file extension.
+     *                              If not given, the extension is extracted from `$path`.
+     *
+     * @return bool
+     */
+    public static function isAllowedMimeType($path, $filename = null)
+    {
+        $whitelist = rex_addon::get('mediapool')->getProperty('allowed_mime_types');
+
+        if (!$whitelist) {
+            return true;
+        }
+
+        $extension = mb_strtolower(rex_file::extension($filename ?: $path));
+
+        if (!isset($whitelist[$extension])) {
+            return false;
+        }
+
+        $mime_type = rex_file::mimeType($path);
+
+        return in_array($mime_type, $whitelist[$extension]);
+    }
+
+    /**
+     * get whitelist of mediatypes(extensions) given via media widget "types" param.
+     *
+     * @param array $args widget params
+     *
+     * @return array whitelisted extensions
+     */
+    public static function getMediaTypeWhitelist($args = [])
```

Comment:
Wenn wir hier schon verschieben, würde ich die begriffe "whitelist" und "blacklist" gerne loswerden (dort wo problemlos möglich).
War ja irgendwann im letzten Jahr mal eine größere Diskussion in der Tech-Szene, dass die Begriffe wohl problematisch sind.
Ob man da nun zustimmt oder nicht, ich denke man sollte sie einfach vermeiden, dort wo es geht. Und eben gerade, wenn wir sowieso schon neue Methoden einführen.

Reply:
das stimmt. bin ich bei dir und passe das an